strax/storage/common.py

```diff
@@ -429,13 +429,6 @@ class StorageBackend:
 
     """
 
-    def __new__(cls, *args, **kwargs):
-        """Mandatorily wrap _read_chunk in a check_chunk_n decorator."""
-        if "_read_chunk" in cls.__dict__:
-            method = getattr(cls, "_read_chunk")
```

Comment:
Thanks @JelleAalbers for the improvement. I understand the request from Carlo and your new implementation. I implemented #758 to force-check the chunk size. Can we do something like `hasattr(cls._read_chunk, '__closure__') and cls._read_chunk.__closure__ is not None` to test whether `_read_chunk` is decorated?

Or use some method like https://stackoverflow.com/a/68583930.

Reply:
Sure, I understand the check is important, and the storage class hierarchy is sufficiently confusing that I understand you reached for decorators. And there will be a way to fix the decorator solution with more metaprogramming (at the risk of making more hard-to-spot bugs).

I think `_read_and_format_chunk` is the better place to do this check. The `_read_chunk` method returns a numpy array, not a strax Chunk: it's `_read_and_format_chunk`'s job to turn that into a chunk. Creating a chunk triggers checks, like whether the data matches the time range, the start and end time are integers, etc. So it makes sense to check the chunk length matches the metadata at the same time.

There is no code that calls  `_read_chunk` directly -- at least  not in the AxFoundation or XENONnT GitHub. The only reason I'd write something like that is to _avoid_ all the checks that come with making a strax chunk, like when trying to read data whose metadata is corrupted. Then you don't want the chunk length check either.